### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Ouail-01**
> ### In this pull request, I will add the logic needed to make the Calculator actually work.

# Learning objectives :
- Handle events in a React app.
- Use React life cycle methods.
- Understand the mechanism of lifting state up.

### CODE REVIEW

**Review by Kamba56 — COMMENTED** [review]
Hi @Ouail-01

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that **Does not have a professional readme**. Please customize the readme for this project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,36 +1,58 @@
...
-        <button type="button" className="or-color">=</button>
+class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
```
Comment (Qoosim) [inline src/components/Calculator.js:12]: - [ ] It is required to make your Calculator component to be functional component by making use of `useState && props`. Kindly check the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md) for this reason. :point_down: 

project-rqts

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,36 +1,58 @@
...
+      this.setState({
+        total: e.target.innerText,
+      });
+    }
+
+    render() {
+      const { total, next, operation } = this.state;
+      return (
+        <div className="calculator-container">
+          <div className="calc-display">{total || next || operation || 0}</div>
```
Comment (Qoosim) [inline src/components/Calculator.js:29]: - [ ] Wonderful job so far. However, there's an issue with your display div. The app behaviour is quite differ from conventional calculator. It only displays first operation performs, but hide other operations to complete ones calculation. It displays correct result when the `=` symbol is clicked to complete the operation though. But every event should reflect on the screen part of the calculation for better user experience. Kindly look into that and fix. Take a look :point_down: 

[app-behavior.webm](https://user-images.githubusercontent.com/26597432/219057092-ef126f10-9349-4bcb-be3a-d1826ba6df36.webm)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Ouail-01, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @Ouail-01 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**